x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -234,13 +241,29 @@ const AssistantComponent: React.FC<Props> = ({
   ////
   //
 
-  const selectedSystemPrompt = useMemo(() => {
-    if (currentConversation.apiConfig.defaultSystemPromptId) {
-      return allSystemPrompts.find(
-        (prompt) => prompt.id === currentConversation.apiConfig.defaultSystemPromptId
+  const selectedSystemPrompt = useMemo(
+    () => getDefaultSystemPrompt({ allSystemPrompts, conversation: currentConversation }),
+    [allSystemPrompts, currentConversation]
+  );
+
+  const [editingSystemPromptId, setEditingSystemPromptId] = useState<string | undefined>(
+    selectedSystemPrompt?.id
+  );
+
+  const handleOnConversationSelected = useCallback(
+    (cId: string) => {
+      setSelectedConversationId(cId);
+      setEditingSystemPromptId(
+        getDefaultSystemPrompt({ allSystemPrompts, conversation: conversations[cId] })?.id
```

Comment:
🪄 healing function ensures the newly-selected conversation has a default system prompt (when applicable)